costmap_2d/include/costmap_2d/costmap_2d_ros.h

```diff
@@ -241,7 +241,10 @@ class Costmap2DROS
   void readFootprintFromConfig(const costmap_2d::Costmap2DConfig &new_config,
                                const costmap_2d::Costmap2DConfig &old_config);
 
-  void resetOldParameters(ros::NodeHandle& nh);
+  void loadOldParameters(ros::NodeHandle& nh);
+  void warnForOldParameters(ros::NodeHandle& nh);
+  void checkOldParam(ros::NodeHandle& nh, const std::string &param_name);
+  void copyParentParameters(const std::string& plugin_name, const std::string& plugin_type, ros::NodeHandle& nh);
```

Comment:
Do you think any of the functionality could be aided by the new functions [here](https://github.com/ros-planning/navigation/blob/melodic-devel/nav_core/include/nav_core/parameter_magic.h)

Reply:
Yes! We could use if for map_type and static_map in checkOldParam. I would suggest to keep copying from the parent namespace to ensure backward compatibility with the old way of defining properties of filters and to give the user the possibility to set for example a default inflation radius for all inflation filters.